### ISSUE TITLE
Autogenerate a batch of model runs, one for each ensemble realisation

### ISSUE THREAD

**Issue by willu47** [nismod/smif#363]
Child of #353 

After successfully configuring the scenarios included within a model run configuration with one variant per ensemble replicate (see #362), a script should automatically generate the required model run configuration files which can subsequently be run as a batch (see #179). 

Using a model run configuration as a template e.g.

```yaml
name: energy_central_001
description:  Energy demand under central scenario, only 2010
stamp: "2017-09-18T12:53:23+00:00"
timesteps:
- 2010
- 2015
- 2020
sos_model: energy
scenarios:
  population: population_med
  water_sector_energy_demand: central
  weather: replicate_001
narratives: {}
<snip>
```

Generate all the model runs required to run the replicates required for the ensemble scenario.

One approach could be to use a placeholder in the template model run for the replicate name, which is then pasted in using the Python [string.Template ](https://docs.python.org/3.7/library/string.html#template-strings) class.  However, this detail should be deferred to the config store - at the store level, the functionality must be generic.

**Comment by willu47** [comment]
@tlestang - looks good. Please could you pad the replicate names with zeros so that they are easier to sort?

**Comment by tomalrussell** [comment]
Closed by #381
